datalad/interface/vcs.py

```diff
@@ -0,0 +1,193 @@
+        # actually new instance. This is unnecessarily costly.
+        try:
+            self._repo = repo_from_path(self.path)
+        except ValueError:
+            lgr.log(5, "Failed to detect a valid repo at %s", self.path)
+            self._repo = None
+            return
+
+        if due.active:
+            # TODO: Figure out, when exactly this is needed. Don't think it
+            #       makes sense to do this for every dataset,
+            #       no matter what => we want .repo to be as cheap as it gets.
+            # Makes sense only on installed dataset - @never_fail'ed
+            duecredit_dataset(self)
+
+        return self._repo
+
+    def close(self):
+        """Perform operations which would close any possibly attached processes
+        """
+        repo = self._repo
+        self._repo = None
+        if repo:
+            # might take care about lingering batched processes etc
+            del repo
+
+    @borrowdoc(GitRepo)
+    def call_git(self, args, files=None, expect_stderr=False, expect_fail=False,
+                 read_only=False):
+
+        return self._repository.call_git(args, files=None, expect_stderr=False,
+                                         expect_fail=False, read_only=False)
+
+    @borrowdoc(GitRepo)
+    def call_git_items_(self, args, files=None, expect_stderr=False, sep=None,
+                        read_only=False):
+        return self._repository.call_git_items_(args, files=None,
+                                                expect_stderr=False,
+                                                sep=None, read_only=False)
+
+    @borrowdoc(GitRepo)
+    def call_git_oneline(self, args, files=None, expect_stderr=False,
+                         read_only=False):
+        return self._repository.call_git_oneline(args, files=None,
+                                                 expect_stderr=False,
+                                                 read_only=False)
+
+    @borrowdoc(GitRepo)
+    def call_git_success(self, args, files=None, expect_stderr=False,
+                         read_only=False):
+        return self._repository.call_git_success(args, files=None,
+                                                 expect_stderr=False,
+                                                 read_only=False)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex_records(self, args, files=None):
+
+        return self._repository.call_annex_records(args, files=None)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex(self, args, files=None):
+
+        return self._repository.call_annex(args, files=None)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex_items_(self, args, files=None, sep=None):
+
+        return self._repository.call_annex_items_(args, files=None, sep=None)
+
+    @borrowdoc(AnnexRepo)
+    def call_annex_oneline(self, args, files=None):
+
+        return self._repository.call_annex_oneline(args, files=None)
```

Comment:
All `call_annex_*()` methods would fail if the underlying `_repository` instance points to a `GitRepo`.